module/Photo/src/Photo/Service/AlbumCover.php

```diff
@@ -0,0 +1,158 @@
+        // Make a blank canvas
+        $target = new Imagick();
+        $target->newImage($config['album_cover']['width'], $config['album_cover']['height'], $config['album_cover']['background']);
+        $index = 0;
+        if (count($images) >= $count) {
+            for ($x = 0; $x < $columns; $x++) {
+                for ($y = 0; $y < $rows; $y++) {
+                    $image = $this->resizeCropImage($images[$index], $imageWidth, $imageHeight);
+                    $index++;
+                    $target->compositeImage($image, imagick::COMPOSITE_COPY, ($imageWidth + $config['album_cover']['inner_border']) * $x + $outerBorderX, ($imageHeight + $config['album_cover']['inner_border']) * $y + $outerBorderY);
+                }
+            }
+        }
+        $target->setImageFormat("png");
+        return $target;
+    }
+
+    /**
+     * Specialized function to rezie and crop photos such that they always
+     * fill the full width and height without damaging the aspect ratio of the
+     * photo.
+     * 
+     * @param Imagick $image The Imagick object to be resized and cropped
+     * @param int $width The desired width
+     * @param int $height The desired height
+     * @return Imagick $image
+     */
+    protected function resizeCropImage($image, $width, $height)
+    {
+        $resizeWidth = max($width, floor($image->getImageGeometry()['width'] / ($image->getImageGeometry()['height'] / $height)));
+        $resizeHeight = max($height, floor($image->getImageGeometry()['height'] / ($image->getImageGeometry()['width'] / $width)));
+        $image->resizeImage($resizeWidth, $resizeHeight, Imagick::FILTER_LANCZOS, 1);
+        $cropX = 0;
+        $cropY = 0;
+        if ($width < $image->getImageGeometry()['width']) {
+            $cropX = floor(($image->getImageGeometry()['width'] - $width) / 2);
+        }
+        if ($height < $image->getImageGeometry()['height']) {
+            $cropY = floor(($image->getImageGeometry()['height'] - $height) / 2);
+        }
+        $image->cropImage($width, $height, $cropX, $cropY);
+        //this second resize may not be needed, needs testing.
+        $image->resizeImage($width, $height, Imagick::FILTER_LANCZOS, 1);
+        return $image;
+    }
+
+    /**
+     * Returns the images needed to fill the album cover
+     * 
+     * @param Photo\Model\Album $album
+     */
+    protected function getImages($album, $count)
+    {
+        $mapper = $this->getAlbumMapper();
+        $config = $this->getConfig();
+        $photos = $mapper->getRandomAlbumPhotos($album, $count);
+        if (count($photos) < $count) {
+            //retrieve more photo's from subalbums until we have enough
+            foreach ($mapper->getSubAlbums($album) as $subAlbum) {
+                $needed = $count - count($photos);
+                if ($needed < 1) {
```

Comment:
Deze if is volgensmij niet nodig, behalve voor performance redenen.

Als dat het geval is kan hij weg